Kernel/Syscalls/mmap.cpp

```diff
@@ -363,7 +363,50 @@ KResultOr<int> Process::sys$mprotect(Userspace<void*> addr, size_t size, int pro
         return 0;
     }
 
-    // FIXME: We should also support mprotect() across multiple regions. (#175) (#964)
+    const auto& regions = space().find_regions_intersecting(range_to_mprotect);
+    // FIXME: what do we do if we have a not mapped page in the range?
+    if (regions.size()) {
+        // first check before doing anything
+        for (auto* region : regions) {
+            if (!region->is_mmap())
+                return EPERM;
+            if (!validate_mmap_prot(prot, region->is_stack(), region->vmobject().is_anonymous(), region))
+                return EINVAL;
+            if (region->access() == prot_to_region_access_flags(prot))
+                return 0;
+            if (region->vmobject().is_inode()
+                && !validate_inode_mmap_prot(*this, prot, static_cast<const InodeVMObject&>(region->vmobject()).inode(), region->is_shared())) {
+                return EACCES;
+            }
+        }
+
+        // then do all the other stuff
+        for (auto* old_region : regions) {
```

Comment:
I'm not convinced this is correct.. some MAP_FIXED tests are in order I think.  I Would want to verify that all the edge cases work out, such as:

1) middle of one region into the middle of another region 
2) middle of one region , past a second region , into the middle of a third region
3) beginning of one region into the middle of a second region
4) begining of one region to the end of a second region
5) beginning of one region, through a second region, to the end of a third region
...
etc.

Reply:
Just reading the code again, There is an Assertion error on a full match while carving everything out

Reply:
I've fixed that and may write tests later

Reply:
There is now a test implemented